src/common/varlen_value.cpp

```diff
@@ -34,23 +33,23 @@ VarlenValue::VarlenValue(const char *data, uint32_t len)
   }
 }
 
-VarlenValue::VarlenValue(const std::string &data)
-                            : Value(Type::GetInstance(Type::VARCHAR)) {
-  uint32_t len = data.length();
-  value_.ptr = new char[len + 1 + sizeof(uint32_t)];
+VarlenValue::VarlenValue(const std::string &data, bool binary) :
+    Value(Type::GetInstance(binary ? Type::VARBINARY : Type::VARCHAR)) {
+  uint32_t len = data.length() + (GetTypeId() == Type::VARCHAR);
+  value_.ptr = new char[len + sizeof(uint32_t)];
   PL_ASSERT(value_.ptr != nullptr);
-  (*(uint32_t *) value_.ptr) = len + 1;
+  (*(uint32_t *) value_.ptr) = len;
   char *dest = value_.ptr + sizeof(uint32_t);
-  PL_MEMCPY(dest, data.c_str(), len + 1);
+  PL_MEMCPY(dest, data.c_str(), len);
 }
 
-VarlenValue::VarlenValue(const Varlen *varlen)
-                            : Value(Type::GetInstance(Type::VARCHAR)) {
-  value_.ptr = new char[varlen->GetSize() + sizeof(uint32_t)];
-  (*(uint32_t *) value_.ptr) = varlen->GetSize();
-  char *dest = value_.ptr;
-  PL_MEMCPY(dest, varlen->GetRaw(), varlen->GetSize());
-}
+//VarlenValue::VarlenValue(const Varlen *varlen, bool binary) :
+//    Value(Type::GetInstance(binary ? Type::VARBINARY : Type::VARCHAR)) {
+//  value_.ptr = new char[varlen->GetSize() + sizeof(uint32_t)];
+//  (*(uint32_t *) value_.ptr) = varlen->GetSize();
+//  char *dest = value_.ptr;
+//  PL_MEMCPY(dest, varlen->GetRaw(), varlen->GetSize());
+//}
```

Comment:
Could you remove the unused code instead of commenting them out? I think we can always find them back through git history if we really need them.